src/billing/components/PayAsYouGo/CancellationOverlay.tsx

```diff
@@ -23,11 +30,42 @@ const CancellationOverlay: FC<Props> = ({onHideOverlay}) => {
   const {handleCancelAccount} = useContext(BillingContext)
   const [hasAgreedToTerms, setHasAgreedToTerms] = useState(false)
   const [hasClickedCancel, setHasClickedCancel] = useState(false)
+  const {
+    reason,
+    canContactForFeedback,
+    suggestions,
+    shortSuggestion,
+  } = useContext(CancelServiceContext)
+  const quartzMe = useSelector(getQuartzMe)
+  const org = useSelector(getOrg)
+
+  const sendDetailsToRudderstack = () => {
+    if (
+      !isFlagEnabled('rudderStackReporting') ||
+      !isFlagEnabled('trackCancellations')
+    ) {
+      return
+    }
+
+    const payload = {
+      org: org.id,
+      tier: quartzMe?.accountType,
+      email: quartzMe?.email,
+      alternativeProduct: shortSuggestion,
+      suggestions,
+      reason: VariableItems[reason],
+      canContactForFeedback: canContactForFeedback ? 'true' : 'false',
+    }
+
+    event('Cancel Org Executed', payload)
+    track('CancelOrgExecuted', payload)
+  }
 
   const handleCancelService = () => {
     if (!hasClickedCancel) {
       setHasClickedCancel(true)
     } else {
+      sendDetailsToRudderstack()
       handleCancelAccount()
     }
```

Comment:
Can we clean this up so that it's something like:

```
if (!hasClickedCancel) {
  setHasClickedCancel(true)
 }
 
 if (isFlagEnabled('rudderStackReporting') && isFlagEnabled('trackCancellations')) {
      sendDetailsToRudderstack()
  }
  handleCancelAccount()
```

The reasoning being we're calling a function that won't do anything above in a particular cirucmstance. This makes it easier to understand what the `sendDetailsToRudderstack` function does, since right now it's more of like a maybe-sendDetailsToRudderStack-if-these-conditions-are-met